pacman/operations/rigged_algorithms/isomorph_check.py

```diff
@@ -0,0 +1,50 @@
+"""A short algorithm to check if there is an isomorphism of the placement of
+vertices by two separate placement algorithms. One of the algorithms must
+output to memory placements_copy and
+<param_type>MemoryPlacements2</param_type> in the algorithms_metadata.xml."""
+
+import os
+import logging
+
+
+class IsomorphicChecker(object):
+
+    def check(self, placements, placements_copy):
+        chip_vertices = set()
+        chip_vertices_copy = set()
+        chips = set()
+        correct = True
+
+        # create a list of processors with assigned vertices
+        for x, y, p in placements.get_placed_processors():
+            chips.add((x, y))
+        # create sets of the vertices on each processor and compare them
+        for (x, y) in chips:
+            for p in range(0, 18):
+                if placements.is_processor_occupied(x, y, p):
+                    chip_vertices.add(placements.get_vertex_on_processor(
+                        x, y, p))
+                if placements_copy.is_processor_occupied(x, y, p):
+                    chip_vertices_copy.add(
+                        placements_copy.get_vertex_on_processor(x, y, p))
+        if chip_vertices != chip_vertices_copy:
```

Comment:
No need for additional comment here (going off the "could I explain it in a year's time?" metric)